impl/authorization/authorization.go

```diff
@@ -38,30 +42,37 @@ func New() authorization.Authorization {
 // authorized to carry the given permission. A call is authorized if:
 //  1. userID matches the identity in sctx,
 //  2. or, sctx's identity is authorized to do the action in domainID and appID.
-func (a *authz) IsAuthorized(sctx *authentication.SecurityContext,
+func (a *authz) Authorize(ctx context.Context,
 	domainID, appID, userID string, permission authzpb.Permission) error {
+	sctx, ok := authentication.FromContext(ctx)
+	if !ok {
+		return status.Errorf(codes.Unauthenticated, "Request does not contain a ValidatedSecurity object")
```

Comment:
The other code returns fixed errors but this doesn't. Should this do the same?

Reply:
The other fixed errors weren't really needed. I've removed them now.

Reply:
OK, this possibly addresses my point above.